### DISCUSSION TITLE
When deleting a user model, the associated role and permission records aren't getting deleted

### DISCUSSION THREAD

**Discussion by leafling** [spatie/laravel-permission#2019] [Q&A]
When I delete a user using:
```
$user->delete()
```
or
```
$user->forceDelete()
```
it doesn't delete the roles or permissions associated with the $user.

If I delete the permissions and/or roles from the `$user` directly through detach, prior to deleting the user, the associated permission and role records do get deleted as desired...
```
$user->permissions()->detach();
$user->roles()->detach();
```

Looking at the traits for Permissions and Roles, it looks like this should be handled automatically with `$user->delete()`. Here is a snippet from `/spatie/laravel-permission/src/Traits/HasPermissions.php` which I believe should be causing these related records to be deleted, but it isn't... 
```
    public static function bootHasPermissions()
    {
        static::deleting(function ($model) {
            if (method_exists($model, 'isForceDeleting') && ! $model->isForceDeleting()) {
                return;
            }

            $model->permissions()->detach();
        });
    }
```

Am I missing something?  I can certainly just detach directly prior to deleting the `$user` as I mentioned above, but I'd like it to work as expected and take care of the detaching automatically. Any ideas?

**Comment by erikn69** [comment]
It is working for me, make a failing test for look at
https://github.com/spatie/laravel-permission/blob/6a3ed627cee28a552b5176c172ae0abc5eb30925/tests/HasPermissionsTest.php#L220-L229
Also the associated role and permission never get deleted, only detached
